src/pubsub/src/publisher/client.rs

```diff
@@ -159,6 +157,60 @@ impl Publisher {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct PublisherBuilder {
+    pub(crate) inner: crate::generated::gapic_dataplane::client::Publisher,
+    topic: String,
+    batching_settings: BatchingSettings,
+}
+
+impl PublisherBuilder {
+    /// Creates a new Pub/Sub publisher builder for topic.
+    pub(crate) fn new(
+        client: crate::generated::gapic_dataplane::client::Publisher,
+        topic: String,
+    ) -> Self {
+        Self {
+            inner: client,
+            topic,
+            batching_settings: BatchingSettings::default(),
+        }
+    }
+
+    // Change the message batching settings.
+    pub fn with_batching(mut self, settings: BatchingSettings) -> PublisherBuilder {
+        self.batching_settings = settings;
+        self
+    }
+
+    pub fn build(self) -> Publisher {
+        Publisher {
+            inner: self.inner,
+            topic: self.topic,
+            batching_settings: self.batching_settings,
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+#[non_exhaustive]
+pub struct BatchingSettings {
+    // To turn off batching, set the value of max_messages to 1.
+    pub message_count_threshold: Option<u32>,
+    pub byte_threshold: Option<u32>,
+    pub delay_threshold_ms: Option<std::time::Duration>,
+}
+
+impl std::default::Default for BatchingSettings {
+    fn default() -> Self {
+        Self {
+            message_count_threshold: Some(100u32),
+            byte_threshold: Some(1000u32),
+            delay_threshold_ms: Some(std::time::Duration::from_millis(10)),
+        }
+    }
+}
+
 #[cfg(test)]
 mod tests {
```

Comment:
Existing test needs to be updated to build the builder.

Also tests for:
- unconfigured Builder uses the correct defaults
- configuration works